app/anti_virus_check.py

```diff
@@ -0,0 +1,136 @@
+import collections
+import time
+
+import requests
+from sdc.rabbit.exceptions import QuarantinableError, RetryableError, BadMessageError
+
+from app import create_and_wrap_logger
+from app import settings
+
+logger = create_and_wrap_logger(__name__)
+
+AVResult = collections.namedtuple('AVResult', 'safe ready scan_results')
+
+
+class AntiVirusCheck:
+    def __init__(self, tx_id):
+        self.bound_logger = logger.bind(tx_id=tx_id)
+
+    def send_for_av_scan(self, payload):
+        self.bound_logger.info("Sending for AV check", filename=payload.file_name)
+        data_id = self._send_for_anti_virus_check(payload.file_name, payload.decoded_contents)
+        self.bound_logger.info("Sent for A/V check", data_id=data_id)
+
+        attempts = 0
+        while attempts <= settings.ANTI_VIRUS_MAX_ATTEMPTS:
+            attempts += 1
+            results = self._get_anti_virus_result(data_id)
+            if not results.ready:
+                time.sleep(settings.ANTI_VIRUS_WAIT_TIME)
+            elif not results.safe:
```

Comment:
What does the anti-virus do if the result isn't ready? Is there always a `safe` attribute present on results? If results.ready evaluates false, and there is no `safe` attribute, an AttributeError will get thrown. If the attribute is present, but isn't initialised (i.e. it could be set to `None`) this block will evaluate true and quarantine a potentially viable upload.